charts/ckan/templates/ckan/deployment.yaml

```diff
@@ -0,0 +1,110 @@
+{{- $name := "ckan" -}}
+{{- $claimName := printf "%s-%s" (include "common.names.fullname" $) $name -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ printf "%s-%s" (include "common.names.fullname" $) $name | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: {{ $name }}
+spec:
+  replicas: {{ .Values.ckan.replicas }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: {{ $name }}
+  template:
+    metadata:
+      {{- with .Values.ckan.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels: {{- include "common.labels.matchLabels" . | nindent 8 }}
+        app.kubernetes.io/component: {{ $name }}
+    spec:
+      {{- include "common.images.renderPullSecrets" ( dict "images" (list .Values.ckan.image) "context" $) | indent 6 }}
+      automountServiceAccountToken: false
+      volumes:
+        - name: "ckan"
+          persistentVolumeClaim:
+            claimName: {{ $claimName }}
+      containers:
+        - name: {{ printf "%s-%s" .Chart.Name $name }}
+          env:
```

Comment:
they are now all referenced. however, the passwords etc are required in two secrets, so that automatic generation in the secret would be difficult.

Reply:
Why does that make it difficult? Can't you just create two secrets?

Reply:
there is a secret with the sql scripts for the database configuration and another one with the secrets for the environment variables

Reply:
Ah, you could create a `.sh` script instead of an `.sql` in which you can use the environment variables with a HEREDOC, see https://stackoverflow.com/a/70976611

Reply:
is fully implemented for all secrets throughout the chart